### PR TITLE
runtime: Revert "runtime: use containerd package instead of cri-containerd"

### PR DESCRIPTION
**Pull request by bergwolf**
This reverts commit 76f16fd1a7eb5bfb04fdac0ac60f7cb5bfcb028e to bring
back cri-containerd crioptions parsing so that kata works with older
containerd versions like v1.3.9 and v1.4.6.

Fixes: #2999

### CODE REVIEW

**Review by liubin — APPROVED** [review]

**Review by gnawux — APPROVED** [review]
+1

Congrats the 3000th issue/PR

**Comment by fidencio** [comment]
Adding `needs-backpot` as it has to reach the `stable-2.3` branch.

**Comment by fidencio** [comment]
@bergwolf, b625f62d4b558796967e666a6e6ea926b6deafd0 also has to be reverted.

**Comment by fidencio** [comment]
And 6d55b1bafa01f2ac88d337aebcb3e6c770be8451 (which was part of the same PR as 76f16fd1a7eb5bfb04fdac0ac60f7cb5bfcb028e)

File: src/runtime/pkg/containerd-shim-v2/create.go [hunk 1]
```diff
@@ -27,11 +27,10 @@ import (
 	"github.com/pkg/errors"
 
 	// only register the proto type
+	crioption "github.com/containerd/containerd/pkg/runtimeoptions/v1"
 	_ "github.com/containerd/containerd/runtime/linux/runctypes"
 	_ "github.com/containerd/containerd/runtime/v2/runc/options"
-
-	crioption "github.com/containerd/containerd/pkg/runtimeoptions/v1"
-	oldcrioption "github.com/containerd/containerd/pkg/runtimeoptions/v1"
+	oldcrioption "github.com/containerd/cri-containerd/pkg/api/runtimeoptions/v1"
```
Comment (fidencio) [inline src/runtime/pkg/containerd-shim-v2/create.go:33]: This is the critical part of this PR, which actually allows us to keep backward compat with what was done with cri-containerd.

The backward compat code was introduced as part fcc93b00749b6b71c3e3d7e7fa028d96c3226d0f and the offending commit it had been set to the same as the "new" crioption (thus, breaking the backwards compat).

**Review by fidencio — APPROVED** [review]
lgtm, thanks @bergwolf!

**Pull request merged**